config/config-mp/src/main/java/io/helidon/config/mp/spi/MpConfigSourceProviderWithProfile.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.config.mp.spi;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+/**
+ * Java Service loader interface for MP ConfigSource Providers that adds configuration profile support.
+ */
+public interface MpConfigSourceProviderWithProfile extends ConfigSourceProvider {
+    /**
+     * Returns a list of configuration sources.
+     *
+     * @param classLoader classloader where resource will be retrieved from (will use the
+     *                    current thread's classloader if value is {@code null})
+     * @param profile configuration profile to use
+     *
+     * @return list of config sources
+     */
+    Iterable<ConfigSource> getConfigSources(ClassLoader classLoader, String profile);
```

Comment:
As with any other API in Helidon, the profile MUST NOT be null.

Reply:
implementation now validates profile  must not be null